parlai/agents/bert_ranker/helpers.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+from parlai.core.torch_ranker_agent import TorchRankerAgent
+import torch
+try:
+    from pytorch_pretrained_bert.modeling import BertLayer, BertConfig
+    from pytorch_pretrained_bert import BertAdam
+    from pytorch_pretrained_bert import BertModel  # NOQA
+except ImportError:
+    raise Exception(("BERT rankers needs pytorch-pretrained-BERT installed. \n "
+                     "pip install pytorch-pretrained-bert"))
+from parlai.core.utils import _ellipse
+
+
+def add_common_args(parser):
+    """Add command line arguments for this agent."""
+    TorchRankerAgent.add_cmdline_args(parser)
+    parser = parser.add_argument_group('Bert Ranker Arguments')
+    parser.add_argument('--history-length', type=int, default=5,
+                        help='Number of previous line to keep for inference')
+    parser.add_argument('--num-samples', type=int, default=131800,
+                        help='Number of samples in the task (temporary)')
+    parser.add_argument('--bert-id', type=str, default='bert-base-uncased')
+    parser.add_argument(
+        '--add-transformer-layer',
+        type="bool",
+        default=False,
+        help="Also add a transformer layer on top of Bert")
+    parser.add_argument('--pull-from-layer', type=int, default=-1,
+                        help="Which layer of Bert do we use? Default=-1=last one.")
+    parser.add_argument('--predefined-candidates-path', type=str, default=None,
+                        help="Path to a list of candidates")
+    parser.add_argument('--token-cap', type=int, default=320,
```

Comment:
wut? 320 tokens?! Or is this like max length?

Would a power of 2-ish number be better? Would TA's `--truncate` do different?

Reply:
Yeah it's a max length. No difference with truncate. 
320 was around the max I could stuff on one GPU for a batch size of 20, so no particular reason. Moved it to truncate.